docs/dub-reference/subpackages.md

```diff
@@ -6,7 +6,130 @@
 
     Tracking issue: <https://github.com/dlang/dub-docs/issues/71>
 
-For adding sub-packages from the registry as a dependency, execute `dub add <packagename>:<subpackage>`
+## Introduction
 
-If you need to compile a sub-package only, use `dub build :<subpackage>`
+DUB has support for sub-packages. They can be used to separate code from
+your root package into dedicated, smaller packages.
 
+Sub-packages can be used individually within the same project, rather than
+needing to maintain and publish multiple packages on a registry.
+All sub-packages share the same versioning as the root package, acting
+like regular individual packages on their own.
+
+For example, using sub-packages to separate client and server code is
+generally a good idea, as it allows separate package settings, such as
+dedicated build types and configurations (as specific tweaks).
+
+!!! note
+
+    Sub-packages can not be easily found in registries, as they are
+    expected to only be used within your project.
```

Comment:
wrong! Sub-packages explicitly allow for usage on the registry.

They just currently have the problem that they are not easily found, but using them for this purpose is still a good idea (e.g. you can use vibe-d:http or serve-d:lsp on their own without needing to depend on all of the project things)

However maintainers might more easily add/remove subpackages, making it more likely that things can suddenly change here.